providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -1,83 +1,43 @@
-import dev.openfeature.flagd.grpc.ServiceGrpc;
-import dev.openfeature.flagd.grpc.ServiceGrpc.ServiceBlockingStub;
-import dev.openfeature.flagd.grpc.ServiceGrpc.ServiceStub;
 import dev.openfeature.sdk.EvaluationContext;
 import dev.openfeature.sdk.EventProvider;
 import dev.openfeature.sdk.FeatureProvider;
 import dev.openfeature.sdk.Metadata;
-import dev.openfeature.sdk.MutableStructure;
 import dev.openfeature.sdk.ProviderEvaluation;
 import dev.openfeature.sdk.ProviderEventDetails;
 import dev.openfeature.sdk.ProviderState;
 import dev.openfeature.sdk.Value;
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import io.grpc.ManagedChannel;
-import io.grpc.Status;
-import io.grpc.StatusRuntimeException;
-import io.grpc.netty.GrpcSslContexts;
-import io.grpc.netty.NettyChannelBuilder;
-import io.grpc.stub.StreamObserver;
-import io.netty.channel.epoll.EpollDomainSocketChannel;
-import io.netty.channel.epoll.EpollEventLoopGroup;
-import io.netty.channel.unix.DomainSocketAddress;
-import io.netty.handler.ssl.SslContextBuilder;
 import lombok.extern.slf4j.Slf4j;
 
-import javax.net.ssl.SSLException;
-import java.io.File;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
-import java.util.function.Function;
-import java.util.stream.Collectors;
-
-import static dev.openfeature.contrib.providers.flagd.Config.BASE_EVENT_STREAM_RETRY_BACKOFF_MS;
-import static dev.openfeature.contrib.providers.flagd.Config.CACHED_REASON;
-import static dev.openfeature.contrib.providers.flagd.Config.CONTEXT_FIELD;
-import static dev.openfeature.contrib.providers.flagd.Config.DEFAULT_DEADLINE;
-import static dev.openfeature.contrib.providers.flagd.Config.FLAG_KEY_FIELD;
-import static dev.openfeature.contrib.providers.flagd.Config.REASON_FIELD;
-import static dev.openfeature.contrib.providers.flagd.Config.STATIC_REASON;
-import static dev.openfeature.contrib.providers.flagd.Config.VALUE_FIELD;
-import static dev.openfeature.contrib.providers.flagd.Config.VARIANT_FIELD;
 
 /**
  * OpenFeature provider for flagd.
  */
 @Slf4j
 @SuppressWarnings("PMD.TooManyStaticImports")
-public class FlagdProvider extends EventProvider implements FeatureProvider, EventStreamCallback {
+public class FlagdProvider extends EventProvider implements FeatureProvider {
     private static final String FLAGD_PROVIDER = "flagD Provider";
 
     private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
-    private ServiceBlockingStub serviceBlockingStub;
-    private ServiceStub serviceStub;
-    private ManagedChannel channel;
-    private final int maxEventStreamRetries;
-    private final Object eventStreamAliveSync;
     private final FlagdCache cache;
     private final ResolveStrategy strategy;
+    private final GrpcConnector grpc;
```

Comment:
Another nit, not a requirement, but in JS we created a generic abstraction (`service`), which `grpc-service` implements. It's a nice way to keep anything explicitly `grpc` related conceptually out of the provider: https://github.com/open-feature/js-sdk-contrib/tree/main/libs/providers/flagd/src/lib/service

Reply:
Added to https://github.com/open-feature/java-sdk-contrib/issues/390